Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -121,57 +129,55 @@ void Digitizer::flush(DigitContainer& digits, o2::dataformats::MCTruthContainer<
 
 SignalContainer Digitizer::addSignalsFromPileup()
 {
-  int count = 0;
   SignalContainer addedSignalsMap;
+  int nSignalsToRemove = 0;
   for (const auto& collection : mPileupSignals) {
+    bool pileupSignalBecomesObsolete = false;
     for (int det = 0; det < MAXCHAMBER; ++det) {
       const auto& signalMap = collection[det]; //--> a map with active pads only for this chamber
       for (const auto& signal : signalMap) {   // loop over active pads only, if there is any
         const int& key = signal.first;
         const SignalArray& signalArray = signal.second;
         // check if the signal is from a previous event
         if (signalArray.firstTBtime < mCurrentTriggerTime) {
-          if ((mCurrentTriggerTime - signalArray.firstTBtime) < BUSY_TIME) {
+          pileupSignalBecomesObsolete = true;
+          if ((mCurrentTriggerTime - signalArray.firstTBtime) > READOUT_TIME) { // OS: READOUT_TIME should actually be drift time (we want to ignore signals which don't contribute signal anymore at mCurrentTriggerTime)
```

Comment:
We want to accumulate all signals which can contribute to the trigger received at `mCurrentTriggerTime`. Each signal has ADC information for 30 time bins (which corresponds to `READOUT_TIME` which is 3 us). So in case the pileup signal is from a previous interaction it can only contribute signal if `signalArray.firstTBtime` (which stores the time of the first time bin for the pileup collision) is closer than `READOUT_TIME` to `mCurrentTriggerTime`. I don't see how the `DEAD_TIME` should play a role here. Or do I miss something? Do we generate a signal for more than 30 time bins for a hit?